docs/user-guide/install-zowe-pswi-deployment.md

```diff
@@ -64,3 +65,28 @@ Before installing, ensure the [z/OSMF requirements](install-zowe-pswi-address-re
         **Note:** You do not have to execute all workflows during PSWI provisioning in z/OSMF immediately.
 
 The deployment process is complete. The new software instance is defined to z/OSMF. You are now ready to Import Product Information into z/OSMF before you install product maintenance.
+
+## Recommendations
+
+### Cleanup
+
+If the job execution fails, the deployment process does not perform a full cleanup of the datasets, before attempting it again, make sure the following datasets are deleted:
+
+```plaintext
+{TARGET_HLQ}.D.AZWEAUTH.#
+{TARGET_HLQ}.D.AZWEZFS.#
+{TARGET_HLQ}.SMPLOG.#
+{TARGET_HLQ}.SMPLOGA.#
+{TARGET_HLQ}.SMPLTS.#
+{TARGET_HLQ}.SMPMTS.#
+{TARGET_HLQ}.SMPSCDS.#
+{TARGET_HLQ}.T.SZWEAUTH.#
+{TARGET_HLQ}.T.SZWEEXEC.#
+{TARGET_HLQ}.T.SZWELOAD.#
+
+{TARGET_HLQ}.CSI.#
+```
```

Comment:
```suggestion
If the job execution fails, the deployment process does not perform a full cleanup of the datasets, before attempting it again, make sure the following datasets are deleted:

```plaintext
{TARGET_HLQ}.PRODUCT.ZOWE.V2.CSI
{TARGET_HLQ}.PRODUCT.ZOWE.V2.D.AZWEAUTH.#
{TARGET_HLQ}.PRODUCT.ZOWE.V2.D.AZWESAMP.#
{TARGET_HLQ}.PRODUCT.ZOWE.V2.D.AZWEZFS.#
{TARGET_HLQ}.PRODUCT.ZOWE.V2.SMPLOG.#
{TARGET_HLQ}.PRODUCT.ZOWE.V2.SMPLOGA.#
{TARGET_HLQ}.PRODUCT.ZOWE.V2.SMPLTS.#
{TARGET_HLQ}.PRODUCT.ZOWE.V2.SMPMTS.#
{TARGET_HLQ}.PRODUCT.ZOWE.V2.SMPPTS.#
{TARGET_HLQ}.PRODUCT.ZOWE.V2.SMPSCDS.#
{TARGET_HLQ}.PRODUCT.ZOWE.V2.SMPSTS.#
{TARGET_HLQ}.PRODUCT.ZOWE.V2.T.SZWEAUTH.#
{TARGET_HLQ}.PRODUCT.ZOWE.V2.T.SZWEEXEC.#
{TARGET_HLQ}.PRODUCT.ZOWE.V2.T.SZWELOAD.#
{TARGET_HLQ}.PRODUCT.ZOWE.V2.T.SZWESAMP.#
{TARGET_HLQ}.PRODUCT.ZOWE.V2.WORKFLOW.#
{TARGET_HLQ}.PRODUCT.ZOWE.V2.ZFS.#
```

If deployment options must be modified and the JCL recreated, before doing so, check the original JCL for the temporary ZFS dataset name like `<user>.SWDEPL.<random_id>.ZFS` and delete it manually. The deployment job will be unable to clean it once it is recreated because the random id will change. 
```